packages/mobile/src/components/drawer/Drawer.tsx

```diff
@@ -506,30 +580,35 @@ export const Drawer: DrawerComponent = ({
      * When the user releases their hold of the drawer
      */
     onPanResponderRelease: (e, gestureState) => {
-      if (isOpen || isOpenToInitialOffset) {
+      if (
+        isOpen ||
+        isOpenIntent.current ||
+        (!isOpen && shouldCloseToInitialOffset)
+      ) {
         // Close if open & drag is past cutoff
         if (
           gestureState.vy > 0 &&
-          gestureState.moveY > height - MOVE_CUTOFF_CLOSE * drawerHeight
+          gestureState.moveY >
+            FULL_DRAWER_HEIGHT - MOVE_CUTOFF_CLOSE * drawerHeight
         ) {
-          if (isOpenToInitialOffset) {
-            slideOut(initialOffsetOpenPosition)
+          if (shouldCloseToInitialOffset) {
+            slideOut(initialOffsetOpenPosition, gestureState.vy, onClose)
```

Comment:
ah so we put onClose here to avoid a state change till after the animation?

Reply:
yee